Sources/Public/ItemViews/SwiftUIWrapperView.swift

```diff
@@ -188,35 +204,48 @@ extension UIResponder {
   }
 }
 
-// MARK: - SwiftUIWrapperView.HostingController
+// MARK: - IDWrapperView
 
+/// A wrapper view that uses the `id(_:)` modifier on the wrapped view so that each one has its own identity, even if it was reused.
 @available(iOS 13.0, *)
-extension SwiftUIWrapperView {
+private struct IDWrapperView<Content: View>: View {
 
-  /// The `UIHostingController` type used by `SwiftUIWrapperView` to embed SwiftUI views in a UIKit view hierarchy.
-  public final class HostingController<Content: View>: UIHostingController<Content> {
+  let content: Content
+  let id: AnyHashable
 
-    // MARK: Lifecycle
+  var body: some View {
+    content
+      .id(id)
+  }
 
-    fileprivate override init(rootView: Content) {
-      super.init(rootView: rootView)
+}
 
-      // This prevents the safe area from affecting layout.
-      _disableSafeArea = true
-    }
+// MARK: - HostingController
 
-    @MainActor required dynamic init?(coder aDecoder: NSCoder) {
-      fatalError("init(coder:) has not been implemented")
-    }
+/// The `UIHostingController` type used by `SwiftUIWrapperView` to embed SwiftUI views in a UIKit view hierarchy. This
+/// exists to disable safe area insets and set the background color to clear.
+@available(iOS 13.0, *)
+private final class HostingController<Content: View>: UIHostingController<Content> {
 
-    public override func viewDidLoad() {
-      super.viewDidLoad()
+  // MARK: Lifecycle
 
-      // Override the default `.systemBackground` color since `CalendarView` subviews should be
-      // clear.
-      view.backgroundColor = .clear
-    }
+  override init(rootView: Content) {
+    super.init(rootView: rootView)
+
+    // This prevents the safe area from affecting layout.
+    _disableSafeArea = true
+  }
+
+  @MainActor required dynamic init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
 
+    // Override the default `.systemBackground` color since `CalendarView` subviews should be
+    // clear.
+    view.backgroundColor = .clear
```

Comment:
unchanged, but I pulled it out of the `SwiftUIWrapperView` extension since it was private anyways.